src/components/TodoContext.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { createContext, useState, useCallback, useMemo } from "react";
+import { useMutation, useQuery } from "@tanstack/react-query";
+interface Todo {
+  id: string;
+  title: string;
+  completed: boolean;
+}
+
+interface TodoContextType {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  onAdd: (newTodo: Todo) => void;
+  onDelete: (todoId: string) => void;
+  handleClearAll: () => void;
+  handleCompleteAll: () => void;
+  getTodos: () => void;
+}
+
+export const TodoContext = createContext<TodoContextType | undefined>(
+  undefined
+);
+
+export const TodoProvider = ({ children }: { children: React.ReactNode }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const TODO_API_URL: string = "https://jsonplaceholder.typicode.com/todos/";
+  const TODOS_NUMBER: number = 5;
+  const TODO_API_URL_LIMIT: string = `https://jsonplaceholder.typicode.com/todos?_limit=${TODOS_NUMBER}`;
+
+  const updateLocalStorage = (todos: Todo[]) => {
+    localStorage.setItem("todos", JSON.stringify(todos));
+  };
+  const { mutate: onAdd } = useMutation({
+    mutationFn: async (newTodo) => {
+      const response = await fetch(TODO_API_URL, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newTodo),
+      });
+
+      if (!response.ok) {
+        throw new Error("Failed to add TODO");
+      }
+
+      const data = await response.json();
+      return data;
+    },
+    onSuccess: (data) => {
```

Comment:
`data` parameter doesn't have a type defined
There are a  few more methods below that have parameters without a type defined.

Reply:
done